java/flight/src/main/java/org/apache/arrow/flight/FlightClient.java

```diff
@@ -77,11 +81,21 @@
   /**
    * Create a Flight client from an allocator and a gRPC channel.
    */
-  private FlightClient(BufferAllocator incomingAllocator, ManagedChannel channel) {
+  private FlightClient(BufferAllocator incomingAllocator, ManagedChannel channel,
+      List<FlightClientMiddleware.Factory> middleware) {
     this.allocator = incomingAllocator.newChildAllocator("flight-client", 0, Long.MAX_VALUE);
     this.channel = channel;
-    blockingStub = FlightServiceGrpc.newBlockingStub(channel).withInterceptors(authInterceptor);
-    asyncStub = FlightServiceGrpc.newStub(channel).withInterceptors(authInterceptor);
+    final ClientInterceptor[] interceptors;
+
+    if (middleware.isEmpty()) {
+      interceptors = new ClientInterceptor[]{authInterceptor};
+    } else {
+      interceptors = new ClientInterceptor[]{authInterceptor, new ClientInterceptorAdapter(middleware)};
+    }
+
+    blockingStub = FlightServiceGrpc.newBlockingStub(channel).withInterceptors(interceptors);
+    asyncStub = FlightServiceGrpc.newStub(channel).withInterceptors(interceptors);
+    // TODO: doGet/doPut need to run all interceptors
```

Comment:
can you make a JIRA describing this issue more clearly?

Reply:
I've fixed it (DoGet/DoPut used to manually run the gRPC interceptor chain, I've refactored that).